service/awsconfig/v10/resource/cloudformation/create.go

```diff
@@ -123,7 +140,15 @@ func (r *Resource) createHostPreStack(ctx context.Context, customObject v1alpha1
 	r.logger.LogCtx(ctx, "level", "debug", "message", "creating the host cluster pre cloud formation stack")
 
 	_, err = r.hostClients.CloudFormation.CreateStack(createStack)
-	if err != nil {
+	if IsAlreadyExists(err) {
+		// TODO this here indicates we should have dedicated resources for the pre,
+		// main and post stacks. The workflow would be more straight forward and
+		// easy to manage. Right now we hack around this and add conditionals to
+		// make it work somehow while bypassing the framework primitives.
+		r.logger.LogCtx(ctx, "level", "debug", "message", "the host cluster pre cloud formation stack is already created")
```

Comment:
Agreed. Splitting this into multiple resources would improve the flow. We already have the adapter and cloudformation services which will help with splitting it.